src/main/java/org/semver4j/internal/range/processor/CaretProcessor.java

```diff
@@ -22,10 +24,12 @@
  * </ul>
  */
 public class CaretProcessor implements Processor {
+    @NotNull
```

Comment:
That's not necessary in favor of field object assignment.

Reply:
Reasons 1 & 5